### PR TITLE
Multiplayer robot selection

### PR DESCRIPTION
**Pull request by BrandonPacewic**
~~Blocked by #869.~~

### Description

Implement client robot selection during connection to multiplayer lobby. Also added a basic outline for what the server and client modes are going to look like.

Not my best code and I would be in favor of re-writing it later but it's a good outline for what multiplayer should look like.

### Objectives
- [x] Create test mode for starting an empty server.
- [x] Create test mode for connecting to an empty server.
- [x] Implement connected client downloading server side robots.
- [x] Create menu for robot selection.
- [x] Move away from testing modes.

### To Test

- Build the project and run it.
- Open up a server in either the unity editor or the built application.
- Then connect to it in either the unity editor or the built application.
- Send some robot data and pick a robot.
- Check out the list of connected clients.

[JIRA Issue - Client Upload Robot Data](https://jira.autodesk.com/browse/AARD-1561)
[JIRA Issue - Multiplayer Robot Selection](https://jira.autodesk.com/browse/AARD-1516)

### CODE REVIEW

File: engine/Assets/Scripts/Modes/Aether/ConnectToMultiplayerMode.cs [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+    public Task<Result<LobbyMessage?, Exception>> SelectRobot(string pathToRobotMira) {
+        if (!File.Exists(pathToRobotMira)) {
+            Logger.Log("Robot does not exist", LogLevel.Error);
+            return Task.FromResult(new Result<LobbyMessage?, Exception>(new Exception("Robot file does not exist.")));
+        }
+
+        var robot = new DataRobot {
+            Name = Path.GetFileNameWithoutExtension(pathToRobotMira),
+            Data = ByteString.CopyFrom(File.ReadAllBytes(pathToRobotMira)),
+            // Guid = _client.Guid
```
Comment (BrandonPacewic) [inline engine/Assets/Scripts/Modes/Aether/ConnectToMultiplayerMode.cs:58]: I had a problem with casting from `ulong` from `ulong?` if someone wants to take a look.

**Review by PepperLola — APPROVED** [review]
Looks good! Everything works properly and we were able to test between two computers.

File: engine/Assets/Scripts/Modes/Aether/ConnectToMultiplayerMode.cs (hunk 1 above)
Comment (PepperLola) [reply engine/Assets/Scripts/Modes/Aether/ConnectToMultiplayerMode.cs:58]: Underneath the assignment you can test if `_client.Guid.HasValue` and then assign `robot.Guid = _client.Guid.Value` if it does.

File: .gitignore [hunk 2]
```diff
@@ -2,6 +2,7 @@
 .vscode/
 .DS_Store
 **/*/.idea/
+/build/
```
Comment (KyroVibe) [inline .gitignore:5]: Whats this for?

**Review by KyroVibe — APPROVED** [review]
Looks good. I'll do a bunch of integration next sprint.

File: .gitignore (hunk 2 above)
Comment (BrandonPacewic) [reply .gitignore:5]: @KyroVibe This is just for if you want to build the project into a build folder. I we have this inside of `engine/` I just added it to the root of the project.

**Pull request merged**